src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -89,16 +93,24 @@ private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progre
                     StartedOrStopped(started: true);
```

Comment:
I see StartedOrStopped now also calls ResetProgressStatus that also seems to set these fields so there might be more races too. We might have to do a larger rework here.

Reply:
I agree - but those other fields are just used to display data (number of items being worked on, status).  I don't think races here would be too impactful (but correct me if I'm wrong).  It could just slightly change the number of items / progress to whichever comes last.

Reply:
Regarding the larger rework, that seems fine - but I don't know that I have time as we're snapping tomorrow, unless we want to go through QB.

Reply:
I'd feel more comfortable making sure we get it right than trying to rush something before QB. My motto is "if the bug is bad enough, then it's easy to take through QB".

Reply:
Sure.  I'll start re-working then :)